### DISCUSSION TITLE
Redirect to returnUrl after login

### DISCUSSION THREAD

**Discussion by DiegoMcDipster** [vercel/next.js#33116] [Help]
Hi,

My scenario:
    user clicks a link to: website/game/play/:gameid
    This is a protected route, so using the code below, the user is redirected to '/login'
    After the login, i would like the user to be automatically directed back to: website/game/play/:gameid

In /game/play/[gameid] getServerSideProps I handle protected route like so:

```
  if (error === "The user is not authenticated") {
    return {
      props: {},
      redirect: {
        destination: "/login",
        permanent: false,
      },
    };
  }
```

That works fine, the user is redirected to the login screen.

What i would like is upon valid login for the user to be redirected back to website/game/play/:gameid

How can I make the url (website/game/play/:gameid) available to the login form?

**Comment by sicarius97** [comment]
@DiegoMcDipster Depends entirely on your auth system. I've used a returnUrl before where I pass it as a callback into the login system functions and redirect back to it once login completes. Next-auth has a lot of these capabilities for instance

**Comment by DiegoMcDipster** [comment]
@sicarius97 thanks for the reply. How do you get the returnUrl from the getServerSideProps to your auth system in the first place?  

The [documentation](https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering) says this about the redirect: 
```
 It should match the shape of { destination: string, permanent: boolean }.
```

I don't see where I could pass the returnUrl to my /login and thereby get it to the auth system. An outline of your code would be very helpful.

Regarding the auth provider.  I'm using [AWS Amplify](https://docs.amplify.aws/lib/auth/emailpassword/q/platform/js/)
